flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/rules/physical/batch/DynamicPartitionPruningRuleTest.java

```diff
@@ -479,10 +543,20 @@ public void testDPPFactorWithDimSideJoinKeyChanged() {
 
         String query =
                 "Select * from fact_part join item on fact_part.id = item.id"
-                        + " join sales on fact_part.id = sales.id"
                         + " join (select dim_date_sk + 1 as dim_date_sk, price from dim) dim1"
                         + " on fact_part.fact_date_sk = dim1.dim_date_sk"
                         + " where dim1.price < 500 and dim1.price > 300";
         util.verifyRelPlan(query);
     }
+
```

Comment:
please add a test which join keys are not partition key but are directly reference the face source fields

Reply:
> please add a test which join keys are not partition key but are directly reference the face source fields

Done!